com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeBrickPool.cs

```diff
@@ -548,4 +545,210 @@ internal void Cleanup()
+        static readonly int _PoolDims = Shader.PropertyToID("_PoolDims");
+
+        static readonly int _State0_L0_L1Rx = Shader.PropertyToID("_State0_L0_L1Rx");
+        static readonly int _State0_L1G_L1Ry = Shader.PropertyToID("_State0_L1G_L1Ry");
+        static readonly int _State0_L1B_L1Rz = Shader.PropertyToID("_State0_L1B_L1Rz");
+        static readonly int _State0_L2_0 = Shader.PropertyToID("_State0_L2_0");
+        static readonly int _State0_L2_1 = Shader.PropertyToID("_State0_L2_1");
+        static readonly int _State0_L2_2 = Shader.PropertyToID("_State0_L2_2");
+        static readonly int _State0_L2_3 = Shader.PropertyToID("_State0_L2_3");
+
+        static readonly int _State1_L0_L1Rx = Shader.PropertyToID("_State1_L0_L1Rx");
+        static readonly int _State1_L1G_L1Ry = Shader.PropertyToID("_State1_L1G_L1Ry");
+        static readonly int _State1_L1B_L1Rz = Shader.PropertyToID("_State1_L1B_L1Rz");
+        static readonly int _State1_L2_0 = Shader.PropertyToID("_State1_L2_0");
+        static readonly int _State1_L2_1 = Shader.PropertyToID("_State1_L2_1");
+        static readonly int _State1_L2_2 = Shader.PropertyToID("_State1_L2_2");
+        static readonly int _State1_L2_3 = Shader.PropertyToID("_State1_L2_3");
+
+        static readonly int _Out_L0_L1Rx = Shader.PropertyToID("_Out_L0_L1Rx");
+        static readonly int _Out_L1G_L1Ry = Shader.PropertyToID("_Out_L1G_L1Ry");
+        static readonly int _Out_L1B_L1Rz = Shader.PropertyToID("_Out_L1B_L1Rz");
+        static readonly int _Out_L2_0 = Shader.PropertyToID("_Out_L2_0");
+        static readonly int _Out_L2_1 = Shader.PropertyToID("_Out_L2_1");
+        static readonly int _Out_L2_2 = Shader.PropertyToID("_Out_L2_2");
+        static readonly int _Out_L2_3 = Shader.PropertyToID("_Out_L2_3");
+
+        internal static void Initialize(in ProbeVolumeSystemParameters parameters)
+        {
+            stateBlendShader = parameters.stateBlendShader;
+            stateBlendKernel = stateBlendShader.FindKernel("BlendStates");
+        }
+
+        const uint s_UnmappedChunk = unchecked((uint)-1);
+
+        List<uint> m_IndexMapping;
+        ComputeBuffer m_ChunkMapping;
+        bool m_MappingNeedsReupload = true;
+        int m_MappedChunks = 0;
+
+        ProbeBrickPool m_State0, m_State1;
+        internal int estimatedVMemCost => m_State0.estimatedVMemCost + m_State1.estimatedVMemCost;
+
+        internal int MaxAvailablebrickCount => GetPoolWidth() * GetPoolHeight() * GetPoolHeight() / ProbeBrickPool.kBrickProbeCountTotal;
+
+        internal int GetPoolWidth() { return m_State0.m_Pool.width; }
+        internal int GetPoolHeight() { return m_State0.m_Pool.height; }
+        internal int GetPoolDepth() { return m_State0.m_Pool.depth; }
+
+
+        internal ProbeBrickBlendingPool(ProbeVolumeTextureMemoryBudget memoryBudget, ProbeVolumeSHBands shBands)
+        {
+            m_State0 = new ProbeBrickPool(memoryBudget, shBands);
```

Comment:
Good for this first PR, but let's make a note that probably we want the budget for blending smaller than the one used for the main pool so that the mapping from the enum to the actual size is different.